File: android/app/src/main/java/com/bensbible/app/model/BookGroup.kt
package com.bensbible.app.model

enum class BookGroup(val displayName: String, val indexRange: IntRange) {
    ALL("All", 0..65),
    OLD_TESTAMENT("Old Testament", 0..38),
    NEW_TESTAMENT("New Testament", 39..65),
    LAW("Law", 0..4),
    HISTORY("History", 5..16),
    POETRY("Poetry", 17..21),
    PROPHETS("Prophets", 22..38),
    GOSPELS("Gospels", 39..42),
    ACTS("Acts", 43..43),
    EPISTLES("Epistles", 44..64),
    REVELATION("Revelation", 65..65);

    fun filterBooks(allBooks: List<String>): List<String> {
        val start = indexRange.first.coerceAtLeast(0)
        val end = indexRange.last.coerceAtMost(allBooks.size - 1)
        if (start > end || start >= allBooks.size) return emptyList()
        return allBooks.subList(start, end + 1)
    }

    companion object {
        val pickerSections = listOf(LAW, HISTORY, POETRY, PROPHETS, GOSPELS, ACTS, EPISTLES, REVELATION)
        val searchFilters = entries.toList()
    }
}

File: BensBible2/Models/BookGroup.swift
import Foundation

enum BookGroup: String, CaseIterable, Identifiable {
    case all
    case oldTestament
    case newTestament
    case law
    case history
    case poetry
    case prophets
    case gospels
    case acts
    case epistles
    case revelation

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .all: "All"
        case .oldTestament: "Old Testament"
        case .newTestament: "New Testament"
        case .law: "Law"
        case .history: "History"
        case .poetry: "Poetry"
        case .prophets: "Prophets"
        case .gospels: "Gospels"
        case .acts: "Acts"
        case .epistles: "Epistles"
        case .revelation: "Revelation"
        }
    }

    var bookIndexRange: ClosedRange<Int> {
        switch self {
        case .all: 0...65
        case .oldTestament: 0...38
        case .newTestament: 39...65
        case .law: 0...4
        case .history: 5...16
        case .poetry: 17...21
        case .prophets: 22...38
        case .gospels: 39...42
        case .acts: 43...43
        case .epistles: 44...64
        case .revelation: 65...65
        }
    }

    func filterBooks(from allBooks: [String]) -> [String] {
        let range = bookIndexRange
        let start = max(range.lowerBound, 0)
        let end = min(range.upperBound, allBooks.count - 1)
        guard start <= end, start < allBooks.count else { return [] }
        return Array(allBooks[start...end])
    }

    static let pickerSections: [BookGroup] = [
        .law, .history, .poetry, .prophets, .gospels, .acts, .epistles, .revelation
    ]

    static let searchFilters: [BookGroup] = BookGroup.allCases
}
